contribute/git-commit-message.md

```diff
@@ -0,0 +1,116 @@
+
+Subject line may be prefixed for continuous integration purposes
+ or better project management with a ticket id. The ticket id could 
+ a Github Issue, Rally Id, JIRA Id, etc.,  For example,
+ if you use Rally to track your development, the subject could be
+ "[TA-1234] test(mayactl): add unit tests for cstor volume list"
+
+### Allowed `<type>`
+* **feat**: A new feature
+* **fix**: A bug fix
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code 
+  (white-space, formatting, missing semi-colons, etc)
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **perf**: A code change that improves performance
+* **test**: Adding missing tests
+* **chore**: Changes to the build process or auxiliary tools and libraries
+   such as documentation generation
+
+### Allowed `<scope>`
+Scope could be anything specifying impacted module/package.
+For example: when committing to openebs/maya repo, the scope can be
+- components : mayactl, m-apiserver, spc-watcher, cast, install, util, etc.
+- generic    : compile, travis-ci, etc.
+
+### `<subject>` text
+Subject line should contains succinct description of the change. 
+
+* use imperative, present tense: “change” not “changed” nor “changes”
+* don't capitalize first letter
+* no dot (.) at the end
+
+### Allowed `<meta>` (Optional)
+Additionally, the end of subject-line may contain twitter-inspired markup
+
+* `#wip` - indicate for contributors the feature being implemented is not 
+   complete yet. Should not be included in changelogs (just the last commit 
+   for a feature goes to the changelog).
+* `#nitpick` - the commit does not add useful information. Used when fixing 
+   typos, etc... Should not be included in changelogs.
+
+### Message body
+* just as in `<subject>` use imperative, present tense: “change” not “changed” nor “changes”
+* includes motivation for the change and contrasts with previous behavior
+
+
+### Message footer
+
+#### Breaking changes
+All breaking changes have to be mentioned in footer with the description of the 
+change, justification and migration notes
+
+```
+BREAKING CHANGE: DEFAULT_REPLICA_NODE_SELECTOR will be ignored
+    The support for using this ENV has been removed. 
+
+    To migrate you are required to mention the node selector in StorageClass
+
+    Before: Specified as a string in the m-apiserver ENV as follows:
+
+    - name: DEFAULT_REPLICA_NODE_SELECTOR
+      value: "nodetype=storage"
+
+    After: Specify in the StorageClass
+    
+    apiVersion: storage.k8s.io/v1
+    kind: StorageClass
+    metadata:
+      name: openebs-jiva-nodeselector
+      annotations:
+        cas.openebs.io/config: |
+          - name: ReplicaNodeSelector
+            value: |-
+              nodetype: storage
+
+    User feedback was to have node-selector configurable per StorageClass to
+    allow scheduling volumes based on the storage attached to the nodes. 
+```
+
+#### Referencing issues
```

Comment:
isn't this the same as Ticket in the heading?

Reply:
Could be same, but ticket could be a tracking issue which is either from github or from  external systems. The issues referenced here could be from other repos within OpenEBS. For example, fixing a PR in maya could actually be related to an issue that user created in openebs/openebs.